lib/puppet/file_serving/metadata.rb

```diff
@@ -128,9 +135,9 @@ def to_data_hash
           'type'   => checksum_type,
           'value'  => checksum
         },
-        'type'         => ftype,
-        'destination'  => destination,
-
+        'type'               => ftype,
+        'destination'        => destination,
+        'source_permissions' => source_permissions,
```

Comment:
@nwolfe previously, puppet's ruby implementation of the `file_metadata` endpoint did not include `source_pemissions` when serializing the response. This lead to inconsistent state on the agent if the agent requested `source_permissions` that were different from its default. For example, a 4.0 agent could request `use`, but the deserialized Metadata objects would have an unspecified `source_permissions` value, and the default value in 4 is `ignore`.

We've decided to include `source_permissions` in the response, similar to `checksum_type`. This should not affect compatibility, because the agent doesn't require or use the `source_permissions` value of objects it gets back. But we likely should modify the clojure endpoint to be consistent. Should I file a SERVER ticket?